### PR TITLE
Details and Delete

### PR DESCRIPTION
**Pull request by Casanova569**
Summary:
The trello card is considered approved if you can access for more detail and delete a movie.

Trello Card:
Name: Detalle y Borrado
Link: https://trello.com/c/JtQ6dbxg/9-detalle-y-borrado

### CODE REVIEW

File: src/Controllers/MovieApp.cs [hunk 1]
```diff
@@ -40,5 +36,65 @@ public IActionResult Create(Movie movie)
...
+            if (id != movie.Id)
+            {
+                return NotFound();
+            } 
+            else if (ModelState.IsValid)
+            {
+                UnitOfWork.Movies.Update(movie);
+                UnitOfWork.Complete();
+            }
+            return RedirectPermanent("/MovieApp");
```
Comment (mikolaslorant) [inline src/Controllers/MovieApp.cs]: Use RedirectToAction instead.

File: src/Controllers/MovieApp.cs [hunk 2]
```diff
@@ -40,5 +36,65 @@ public IActionResult Create(Movie movie)
...
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, [Bind("Id,Title,ReleaseDate,Genre,Price")] Movie movie)
+        {
+            if (id != movie.Id)
+            {
+                return NotFound();
+            } 
+            else if (ModelState.IsValid)
```
Comment (mikolaslorant) [inline src/Controllers/MovieApp.cs]: maybe ask if (!ModelState.IsValid) and return an error.

File: src/Repositories/Repository.cs [hunk 3]
```diff
@@ -68,5 +68,10 @@ public void RemoveRange(IEnumerable<TEntity> entities)
         {
             Context.Set<TEntity>().RemoveRange(entities);
         }
+
+        public void Update(TEntity entity)
+        {
+            Context.Set<TEntity>().Update(entity);
+        }
     }
 }
```
Comment (mikolaslorant) [inline src/Repositories/Repository.cs]: add Eof

**Review by mikolaslorant — CHANGES_REQUESTED** [review]

File: src/Controllers/MovieApp.cs [hunk 4]
```diff
@@ -54,16 +56,49 @@ public IActionResult Edit(int id)
         [ValidateAntiForgeryToken]
         public IActionResult Edit(int id, [Bind("Id,Title,ReleaseDate,Genre,Price")] Movie movie)
         {
-            if (id != movie.Id)
+            if (!ModelState.IsValid)
             {
-                return NotFound();
+                return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Watch out here, you are removing the movie id check. you should probably check if the movie exists before getting it. what happens if a non existent id is sent?

File: src/Views/MovieApp/Delete.cshtml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+@using (Html.BeginForm()) {
+    @Html.AntiForgeryToken()
+    <h1>Are you sure?</h1>
```
Comment (truccomarcos) [inline src/Views/MovieApp/Delete.cshtml]: Use the localizer for this string

File: src/Views/MovieApp/Details.cshtml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            <th>
+                @Html.DisplayNameFor(model => model.Genre)
+            </th>
+            <th>
+                @Html.DisplayNameFor(model => model.Price)
+            </th>
+        </tr>
+    </thead>
+    <tbody>
+@if (Model != null)
```
Comment (truccomarcos) [inline src/Views/MovieApp/Details.cshtml]: Should we be able to get to this view if the model was null?

**Review by truccomarcos — CHANGES_REQUESTED** [review]

File: src/Controllers/MovieAppController.cs [hunk 7]
```diff
@@ -71,5 +73,38 @@ public IActionResult Edit(int id, Movie movie)
...
+            if (movie == null)
+            {
+                return NotFound();
+            }
+            return View(movie);
+        }
+
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public IActionResult DeleteConfirmed(int id, bool notUsed)
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: what is `bool notUsed` for?

File: src/Views/MovieApp/Delete.cshtml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+@using Microsoft.AspNetCore.Mvc.Localization
+@using System.Globalization;
+@inject IViewLocalizer Localizer;
+@{
+    ViewData["Question"] = Localizer["DeleteQuestion"];
+}
+@using (Html.BeginForm()) {
+    @Html.AntiForgeryToken()
+    <h1>@ViewData["Question"]</h1>
```
Comment (truccomarcos) [inline src/Views/MovieApp/Delete.cshtml:10]: Does this show the movie being deleted?

**Review by truccomarcos — CHANGES_REQUESTED** [review]

**Review by truccomarcos — APPROVED** [review]

**Pull request merged**